File: isPalindromeLinkedList.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func isPalindrome(head *ListNode) bool {
    index, counter, convertedLL, startIndex, endIndex := head, 0, make([]int, 0), 0, 0
    
    for index != nil{
        convertedLL = append(convertedLL, index.Val)
        counter += 1
        index = index.Next
    }
    
    endIndex = len(convertedLL) - 1
    for endIndex >= startIndex{
        if convertedLL[endIndex] != convertedLL[startIndex]{
            return false
        }
        endIndex -= 1
        startIndex += 1
    }
    return true
}

File: isPalindromeLinkedList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        listToString = ''
        index = 0
        while head != None:
            listToString = listToString + str(head.val)
            head = head.next
            index = index + 1
        revString = listToString[::-1]
        return (revString == listToString)
